### DISCUSSION TITLE
Icon

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#277] [Component suggestions]
<img width="480" alt="verschillende vormen van de icon component" src="https://github.com/user-attachments/assets/a419ec73-b738-49e8-84f5-0bf9d8fc1305">

We willen graag meer leren over de Icon component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1726053300427049)

---

## Naam
Icon

## Doel
Grafisch symbool dat visueel informatie geeft over een actie, onderwerp of status.

## Onderzoek
[Accounts & Self-Service UX: Consider Having an “Icon-Based” Dashboard - Baymard Institute](https://baymard.com/blog/use-icons-in-the-account-dashboard)
Gebruikersonderzoek naar het gebruik van iconen in een dashboard. Eén van de inzichten: door gebruik te maken van iconen (i.c.m. tekst), verbeterde de scanbaarheid aanzienlijk van het dashboard voor deelnemers.

[Iconen - Gebruikersonderzoeken.nl](https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/iconen/utrecht-bevrijdingsdag)
Gebruikersonderzoeken naar iconen door gemeente Utrecht en OpenGemeenten. 

[Usability Testing of Icons - Nielsen Norman Group](https://www.nngroup.com/articles/icon-testing/)
Artikel dat uitlegt welke verschillende soorten onderzoeken je in diverse stadia kunt uitvoeren om ervoor te zorgen dat mensen de betekenis en het doel van iconen begrijpen.

[Improving Icon Usability and Accessibility: 6 Valuable Tips - The A11Y Collective](https://www.a11y-collective.com/blog/icon-usability-and-accessibility/)
Artikel waarin 6 praktische tips staan om je icoonontwerpen te verbeteren van goed naar uitzonderlijk, met de nadruk op inclusiviteit en gebruiksgemak.

[Accessible SVGs - CSS-tricks](https://css-tricks.com/accessible-svgs/)
Artikel dat de impact onderzoekt van deze verandering naar SVG iconen op gebruikers van ondersteunende technologieën (zoals screenreaders) en wat er nodig is om ervoor te zorgen dat SVG’s toegankelijk en bruikbaar zijn voor alle gebruikers.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-media-icon--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=6852-5124&node-type=canvas&t=wao8UE9e1XsiKrAP-11)
- [Github](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/icon)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="63" alt="image" src="https://github.com/user-attachments/assets/989503ff-7d37-4ba6-8c91-7a603ac7bef4">

With text
Icons can be used alongside text. Using the same text levels for both the icon and text aligns them perfectly.
<img width="219" alt="image" src="https://github.com/user-attachments/assets/94b0134d-4f79-4739-8c3d-50cb4f00c41b">

Square
Sometimes it’s useful for the icon to take up square space, for example, with the Icon Button.
<img width="63" alt="image" src="https://github.com/user-attachments/assets/68aa965c-e4cd-4411-a938-222ba76feb60">

Level 3
<img width="60" alt="image" src="https://github.com/user-attachments/assets/6b415479-fcc7-422b-b635-6daf03f8f647">

Level 4
<img width="60" alt="image" src="https://github.com/user-attachments/assets/defe80e1-0313-42f4-8874-6b6565fe65db">

Level 5
<img width="60" alt="image" src="https://github.com/user-attachments/assets/58a28bd9-6afd-44a1-9a6e-7acc1d0def77">

Level 6
<img width="60" alt="image" src="https://github.com/user-attachments/assets/c77b1411-3016-45b7-9d92-be5fbc863e09">

**Figma**
De iconen zijn als placeholders aangegeven in verschillende sizes. 
<img width="667" alt="image" src="https://github.com/user-attachments/assets/9129a362-9238-4b5e-9e88-d13f8e34f419">


---

### Notities

- Format: SVG.
- SVG zit in een `<span>`.
- Kleur van icoon is afhankelijk van de context.
- In Figma heb je de varianten 'filled' en 'line'.
- Beschikbaar in 4 verschillende sizes, die zijn aangeduid in levels.
- Documentatie beschikbaar (Storybook en Figma).

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-data-display-icons--docs)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=3497-14242&node-type=frame&t=7F2YG2x9AIgD09AP-11)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=3-967&node-type=canvas&t=xYKGmrP4nd7Ff5hp-11)
- [Github](https://github.com/nl-design-system/denhaag/tree/main/components/Icons)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="51" alt="image" src="https://github.com/user-attachments/assets/8d23318a-178d-4858-a247-62cd5165c619">

All
<img width="968" alt="image" src="https://github.com/user-attachments/assets/1e9da632-d7e9-4699-8138-d201dc819c2e">

---

### Notities
- Format: SVG.
- Kleur van icoon is afhankelijk van de context.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Iconen hebben allemaal een vaste grootte (24px).
- Tabler iconen in gebruik.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/identity/icon)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/docs/html-css-icon--docs)
- [Github](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/icon)

---

### Screenshots / Varianten

**Storybook**

User
<img width="39" alt="image" src="https://github.com/user-attachments/assets/5807525a-0937-4f74-8194-760f9683f3cc">

Table
<img width="44" alt="image" src="https://github.com/user-attachments/assets/9193f40e-3687-4cbc-94b6-95e55ca06b01">

---

### Notities
- Format: SVG.
- Icoon wordt aangeleverd in zwart. Kleur kan worden bepaald via CSS.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Iconen hebben allemaal een vaste grootte (24px).
- In het `use` element wordt direct `href` in plaats van `xlink:href` gebruikt.
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## DUO

- [Documentatie website](https://uno.dfront.rijkscloud.nl/#/richtlijnen/icons)

---

### Screenshots / Varianten

**Documentatie website**

<img width="51" alt="image" src="https://github.com/user-attachments/assets/dd83ed71-24de-4331-8b2c-b886d0b4d726">

Icon met badge
Een icon kan voorzien worden van een badge om extra info op een icon te tonen door gebruik te maken van het `uno-ng-icon` component.
<img width="54" alt="image" src="https://github.com/user-attachments/assets/8ef1cb0a-edff-499a-906e-87b6bad39b61">

Kleuren
Standaard heeft een icoon de kleur van tekstkleur van het parent element. Daarnaast is het ook mogelijk de kleur van het icoon te definieren, mogelijke kleuren zijn: `Wit` `Zwart` `Blauw`
<img width="122" alt="image" src="https://github.com/user-attachments/assets/374d57e1-25e6-41df-ba51-d966ebf3c52a">

Formaten
Standaard is een icoon `20px` groot. Daarnaast is het ook mogelijk de kleur van het icoon te definieren, mogelijke kleuren zijn: `S (16px)` `M (24px)` `L (32px)` `XL (40px)`
<img width="234" alt="image" src="https://github.com/user-attachments/assets/b2f9767b-3595-4212-89fa-61de6b1706cf">

---

### Notities
- Format: SVG.
- Kleur: beschikbaar in 3 kleuren. Heeft standaard kleur van tekstkleur van parent element.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Beschikbaar in verschillende groottes: 16px, 24px, 32px en 40px.
- Toegankelijkheid: er wordt een `role` aan de icon toegekend, eventueel aangevuld met het `title` attribuut, zodat screenreaders begrijpen waarvoor het element bedoeld is.
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## OpenGemeenten

- [Patternlab](https://patternlab.lenteveld.nl/patterns/00-atoms-02-buttons-00-Button-OpenGemeentenLinkHandling-Button-ArrowStyleChevronAfter/00-atoms-02-buttons-00-Button-OpenGemeentenLinkHandling-Button-ArrowStyleChevronAfter.html)

---

### Screenshots / Varianten

**Patternlab**

Voorbeelden van een aantal componenten waarin een icoon te vinden is.

Button (Arrow After)
<img width="163" alt="image" src="https://github.com/user-attachments/assets/e18e86fc-89e3-46d0-a2bb-486ad85983d1">

Button (Icon Municipality Set)
<img width="219" alt="image" src="https://github.com/user-attachments/assets/eea41e93-1b13-4617-b637-5544254d61bc">

Typolink (File Audio)
<img width="310" alt="image" src="https://github.com/user-attachments/assets/7754b64a-0777-4b57-bce5-10eef6ee6303">

OpenGemeentenSite Metadata All
<img width="646" alt="image" src="https://github.com/user-attachments/assets/132be80e-ae37-4ca4-9ab2-b7754fc45dcd">

Menu Toggle Button
<img width="95" alt="image" src="https://github.com/user-attachments/assets/8a2d1595-52b3-43c3-ad0c-67a74c431893">

Menu Link
<img width="649" alt="image" src="https://github.com/user-attachments/assets/717ad668-2c25-482b-a8a3-8e7db50b1f36">

Top Tasks List
<img width="310" alt="image" src="https://github.com/user-attachments/assets/79e08811-feff-4535-a576-b2b4e092ebe8">

---

### Notities
- Geen afzonderlijke Icon component beschikbaar.
- Format: SVG.
- Kleur van icoon is afhankelijk van de context.
- Beschikbaar in verschillende groottes: 16px, 18px, 20px, 24px, 36px, 42px en 48px. De hoogte en breedte kunnen hierbij variëren.

**Comment by Rozerinay** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/rotterdam-iconen-overzicht--docs)
- [Figma](https://www.figma.com/design/RiVsTfcbmnKSU2BGqQBI9n/RODS-Foundation-OLD?node-id=1-1469)

---

### Screenshots / Varianten


**Storybook**
<img width="47" alt="image" src="https://github.com/user-attachments/assets/555b4149-c83c-4ffc-bbc8-d979fccb6dcb">

**Figma**
<img width="700" alt="image" src="https://github.com/user-attachments/assets/32b271f6-1a2e-4c25-90f8-066de98b95c7">

---

### Notities
- Format: SVG.
- Kleur van icoon is afhankelijk van de context.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Iconen hebben allemaal een vaste grootte (32px).
- Wordt meestal gebruikt in combinatie met tekst, zoals in Buttons en Links.
- Documentatie beschikbaar (Figma).

**Comment by Rozerinay** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/componenten/icon/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-icon--documentatie)
- [Github](https://github.com/nl-design-system/rvo/tree/master/components/icon)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS--RVO-Design-System-?node-id=50-13)

---

### Screenshots / Varianten

**Storybook**

<img width="48" alt="image" src="https://github.com/user-attachments/assets/a3165176-30ed-4645-89f1-54ced5044f36">


---

### Notities
- Format: `span`.
- Role: `img`.
- Kleur: beschikbaar in 5 kleuren.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Beschikbaar in verschillende groottes: xs, sm, md, lg, xl, 2xl, 3xl en 4xl. 
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-icon--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=3-418)
- [Github](https://github.com/nl-design-system/utrecht/tree/main/components/icon)

---

### Screenshots / Varianten

**Storybook**

Unicode
<img width="61" alt="image" src="https://github.com/user-attachments/assets/083af728-ed49-46b3-9fec-342f7d81afdf">

Unicode Emoji
<img width="63" alt="image" src="https://github.com/user-attachments/assets/a716915c-64bc-424a-9e00-bde9b3825691">

---

### Notities
- Format: SVG.
- Kleur van icoon is afhankelijk van de context. Kan kleur van parent element overnemen.
- Filled en outline iconen beschikbaar (niet voor elk icoon afzonderlijk).
- Grootte van icoon is instelbaar. 
- Variable voor grootte en kleur beschikbaar. De hoogte en breedte kunnen hierbij variëren.
- Design tokens beschikbaar.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Icon component op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=icon)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Icon op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=icon)
